artemis/modules/api_scanner.py

```diff
@@ -0,0 +1,168 @@
+                    except Exception as e:
+                        self.log.info(f"Unable to validate spec at {try_url}: {e}")
+                        continue
+            except Exception as e:
+                self.log.debug(f"Error checking {try_url}: {e}")
+                continue
+        return None
+
+    def install_offat(self) -> None:
+        if not os.path.exists("/offat"):
+            os.system("git clone https://github.com/OWASP/OFFAT /offat")
+            patch_file_path = os.path.join(os.path.dirname(__file__), "data/offat", "offat_artemis.patch")
+            os.system(f"git -C /offat apply {patch_file_path}")
+
+        os.system("pip install -e /offat/src")
+
+    def scan(self, target_api_specification: str) -> Dict[str, Any]:
+        output_file = "/tmp/output.json"
+
+        if Config.Modules.APIScanner.ONLY_GET_REQUESTS:
+            os.system(
+                " ".join(
+                    [
+                        "offat",
+                        "-f",
+                        target_api_specification,
+                        "--only-get-requests",
+                        "-o",
+                        output_file,
+                        "--format",
+                        "json",
+                    ]
+                )
+            )
+        else:
+            os.system(" ".join(["offat", "-f", target_api_specification, "-o", output_file, "--format", "json"]))
+
+        with open(output_file) as f:
+            file_contents = f.read()
+
+        report: Dict[str, Any] = json.loads(file_contents)
+        os.unlink(output_file)
+        return report
+
+    def run(self, current_task: Task) -> None:
+        url = get_target_url(current_task)
+
+        # Try to discover the API spec
+        spec_file = self.discover_spec(url)
+        if not spec_file:
+            self.db.save_task_result(
+                task=current_task, status=TaskStatus.OK, status_reason="No OpenAPI/Swagger specification found", data={}
+            )
+            return
+
+        try:
+            results = []
+            test_results = self.scan(spec_file)
+
+            for result in test_results.get("results", {}):
+                if result.get("vulnerable", False):
+                    results.append(
+                        APIResult(
+                            url=result.get("url"),
+                            method=result.get("method"),
+                            vulnerable=result.get("vulnerable"),
+                            details=result.get("vuln_details"),
+                            curl_command=result.get("curl_command"),
+                            status_code=result.get("response_status_code"),
+                        )
+                    )
+
+            if results:
+                status = TaskStatus.INTERESTING
+                status_reason = "Found potential vulnerabilities in the api"
+            else:
+                status = TaskStatus.OK
+                status_reason = None
```

Comment:
even if we don't find anything, I'd save the detected swagger path ("status=ok, reason="detected API on ..., no vulnerasbilities found")